src/main/java/ch/hearc/todont/models/User.java

```diff
@@ -55,7 +45,78 @@
 
     // METHODS
 
-    public User() {}
+    /**
+     * Pledges this user to the given ToDont
+     * 
+     * @param toDont ToDont the user will be pledged to
+     * @return The new pledge object created in this method, or an existing pledge
+     *         to the Todont if it exists
+     */
+    public Pledge pledgeTo(ToDont toDont) {
+        Pledge pledge = getPledgeTo(toDont);
+
+        if (pledge == null) {
+            pledge = new Pledge();
+            pledge.setUser(this);
+            pledge.setToDont(toDont);
+            pledge.setDateJoined(Timestamp.from(Instant.now()));
+        }
+
+        return pledge;
+    }
+
+    /**
+     * Get the pledge made by this user to the given ToDont if it exists
+     * 
+     * @param toDont A ToDont the user should be pledged to
+     * @return The Pledge object, if it exists, else null
+     */
+    public Pledge getPledgeTo(ToDont toDont) {
+        for (Pledge pledge : pledges) {
```

Comment:
Isn't this a "n + 1" problem? Couldn't we just "query" the pledge owned by this user for the todont instead of looping trough everything?

Reply:
Knowing the repositories shouldn't be used in this context, how would one avoid this issue ? I couldn't think of one, granted I didn't spend a very long time searching.

Reply:
Perhaps this method shouldn't exist in the model, and the controllers would have to use the pledge repository.

Reply:
I don't know much either, would it be possible to retrieve the pledge using the repository by giving it a todont and a user?

Reply:
Actually yes. Furthermore, I already implemented it, but then forgot. This method will be deleted in the next push.